components/camera/rtsp/decode.go

```diff
@@ -2,26 +2,146 @@ package rtsp
 
 import (
 	"bytes"
+	"context"
+	"fmt"
 	"image"
-	"image/jpeg"
+	"unsafe"
 
-	"github.com/aler9/gortsplib/v2/pkg/format"
-	"github.com/pion/rtp"
 	"github.com/pkg/errors"
+
+	"go.viam.com/rdk/gostream/ffmpeg"
+	"go.viam.com/rdk/rimage"
+	"go.viam.com/rdk/utils"
 )
 
-type decoder func(pkt *rtp.Packet) (image.Image, error)
+// Decoder the decoder created for the RTSP stream.
+type Decoder struct {
+	codec       ffmpeg.Codec
+	context     ffmpeg.CodecContext
+	fmtCxt      ffmpeg.FormatContext
+	videoStream int
+	cancelCtx   context.Context
+	cancelFunc  func()
+}
+
+// NewDecoder returns a new decoder.
+func NewDecoder(ctx context.Context, addr string) (*Decoder, error) {
+	dec := &Decoder{}
+	dec.fmtCxt = ffmpeg.AllocFormatContext()
+	dec.cancelCtx, dec.cancelFunc = context.WithCancel(context.Background())
+
+	dict, err := ffmpeg.NewDictionary(ctx, ffmpeg.DictionaryEntries{
+		// Use `ffmpeg -h full` for a complete list of options
+		"rtsp_transport": "tcp",
+		"scan_all_pmts":  "1",
+		"sdp_flags":      "rtcp_to_source",
+		"buffer_size":    "-1",
+		"timeout":        "-1",
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot create dictionary")
+	}
+	defer dict.Free()
+
+	if err := dec.fmtCxt.OpenInput(ctx, addr, dict); err != nil {
+		return nil, errors.Wrap(err, fmt.Sprintf("cannot open input for address '%s'", addr))
+	}
+
+	if err := dec.fmtCxt.FindStreamInfo(ctx); err != nil {
+		return nil, errors.Wrap(err, "cannot find stream information")
+	}
```

Comment:
That's my understanding too. With debug logs on you can see the SDP information. I believe this allows it to build that description from the information it finds in the stream. I'm not sure what next steps would be if this fails since I'm not sure how FFmpeg would communicate with RTSP stream it knows nothing about. In general, I don't think we want to surface more options to the user if we can avoid it. Ideally, we'd take in an RTSP address and figure everything else out.